challenge_serving_drinks/src/challenge_serving_drinks/get_order.py

```diff
@@ -0,0 +1,224 @@
+# System
+import random
+
+# ROS
+import rospy
+import smach
+
+# TU/e Robotics
+import robot_smach_states as states
+from hmi import TimeoutException
+from robocup_knowledge import knowledge_loader
+from robot_skills.robot import Robot
+from robot_smach_states.util.designators import EdEntityDesignator
+
+# Knowledge
+COMMON_KNOWLEDGE = knowledge_loader.load_knowledge("common")
+
+
+class AskDrink(smach.State):
+    """ Asks the operator what he/she would like to drink.
+
+    This is based on the 'TakeOrder' class of the the restaurant challenge. Might
+    be nice to merge these two.
+    """
+    def __init__(self, robot, drink_designator, max_tries=3, max_queries_per_try=3):
+        # type (Robot, EdEntityDesignator) -> None
+        """ Initialization method
+
+        :param robot: robot api object
+        :param drink_designator: designator in which the drink (entity type) is stored
+        :param max_tries: (int) maximum number of times the robot asks which drink
+        :param max_queries_per_try: (int) maximum number of queries to the HMI server per try
+        """
+        smach.State.__init__(self, outcomes=["succeeded", "failed"])
+        self._robot = robot
+        self._drink_designator = drink_designator
```

Comment:
The _'true designator way'_ to do this would be to write the drink type to a `type_des = VariableDesignator(resolve_type=str)` and pass that to an `EdEntityDesignator(type=type_des)`. 

This allows you to get the name of the drink in other places directly as well.

Reply:
See 1308fc8fa885d04c9b317d6d0fd3b5eec81f6af0